components/tidb_query_expr/src/impl_vec.rs

```diff
@@ -69,7 +70,7 @@ mod tests {
             (vec![1.0, 2.0, 3.0], Some(3)),
         ];
         for (arg, expected_output) in cases {
-            let arg = VectorFloat32::from_f32(arg).unwrap();
+            let arg = VectorFloat32::copy_from_f32(arg.as_slice());
```

Comment:
A possible simplify:

```suggestion
            let arg = VectorFloat32::copy_from_f32(&arg);
```